src/main/java/trackr/logic/commands/EditTaskCommand.java

```diff
@@ -0,0 +1,187 @@
+package trackr.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static trackr.logic.parser.CliSyntax.PREFIX_DEADLINE;
+import static trackr.logic.parser.CliSyntax.PREFIX_NAME;
+import static trackr.logic.parser.CliSyntax.PREFIX_STATUS;
+import static trackr.model.Model.PREDICATE_SHOW_ALL_TASKS;
+
+import java.util.List;
+import java.util.Optional;
+
+import trackr.commons.core.Messages;
+import trackr.commons.core.index.Index;
+import trackr.commons.util.CollectionUtil;
+import trackr.logic.commands.exceptions.CommandException;
+import trackr.model.Model;
+import trackr.model.task.Task;
+import trackr.model.task.TaskDeadline;
+import trackr.model.task.TaskName;
+import trackr.model.task.TaskStatus;
+
+/**
+ * Edits the details of an existing task in the task list.
+ */
+public class EditTaskCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit_task";
+    public static final String COMMAND_WORD_SHORTCUT = "edit_t";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the task identified "
+            + "by the index number used in the displayed task list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "TASK NAME] "
+            + "[" + PREFIX_DEADLINE + "TASK DEADLINE] "
+            + "[" + PREFIX_STATUS + "TASK STATUS]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "Buy sugar "
+            + PREFIX_STATUS + "D";
+
+    public static final String MESSAGE_EDIT_TASK_SUCCESS = "Edited Task: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_TASK = "This task already exists in the task list.";
+
+    private final Index index;
+    private final EditTaskDescriptor editTaskDescriptor;
+
+    /**
+     * @param index of the task in the filtered task list to edit
+     * @param editTaskDescriptor details to edit the task with
+     */
+    public EditTaskCommand(Index index, EditTaskDescriptor editTaskDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editTaskDescriptor);
```

Comment:
Can be put into a single line
```suggestion
        requireNonNull(index, editTaskDescriptor);
```